### ISSUE TITLE
Split up the monolith

### ISSUE THREAD

**Issue by vieux** [moby/moby#32872]
This repository contains the code of the legacy "docker engine", a big monolith.
Our goal, as part of the moby project is to split this monolith into multiple components.

a **few** examples are: builder, image store, distribution, etc...

A good 1st step would be to have a list of components we aim to get.

**Comment by AkihiroSuda** [comment]
- "swarmd"
- How will "image store" differ from the one embedded in containerd?
(At DockerCon/MobySummit, I think I saw/hear the concept of "layerd" - is it what called "image store" here?)
- IIUC the discussion at the summit, the Dockerfile parser will be moved to docker/cli? What will be the role of "builder"?

**Comment by rogersachan** [comment]
![image](https://cloud.githubusercontent.com/assets/7173984/25468337/4308e2a8-2ae3-11e7-9cc2-a037909ea9c3.png)
This is the image I got in my inbox regarding the docker split. IS this not the planned splitup?

**Comment by vdemeester** [comment]
@rogersachan yes this is most likely what should happen. We need to make sure we didn't miss anything and see how to split the monolith into these pieces. But we should definitely use this as a start for components to split 👼

**Comment by dnephin** [comment]
Once the components are split:
* how will they discover each other (for inter-component communication)?
* what is responsible for starting/stopping/reloading all the components?
* how will component failures be communicated to the user responsible for keeping them running?
* will events (and logs) from different components be aggregated into a single stream?
* how will the client discover which components are available? How will it know the address (tcp or socket) to communicate with a particular component?

**Comment by rogersachan** [comment]
These are the repos in that image:
* https://github.com/containerd/containerd
* https://github.com/opencontainers/runc
* https://github.com/docker/swarmkit
* https://github.com/linuxkit/linuxkit
* https://github.com/docker/infrakit
* https://github.com/docker/notary
* https://github.com/docker/distribution
* https://github.com/docker/libnetwork
* https://github.com/moby/vpnkit
* https://github.com/moby/datakit
* https://github.com/moby/hyperkit
* https://github.com/docker/compose

It should be noted that docker still has ownership of swarmkit, infrakit, notary, distribution, libnetwork, and compose. Are these supposed to be transferred over to moby or will these actually become part of Docker CE/EE?

**Comment by rogersachan** [comment]
I assume this is completed?
